File: model/src/main/java/io/github/runtimemodels/chazm/id/IdModule.java
package io.github.runtimemodels.chazm.id;

import com.google.inject.AbstractModule;

/**
 * The {@linkplain IdModule} class provides a Guice binding module for id.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
public class IdModule extends AbstractModule {

	@Override
	protected void configure() {
		bind(IdFactory.class).to(IdFactoryImpl.class);
	}

}

File: chazm-model/src/main/kotlin/runtimemodels/chazm/model/id/IdModule.kt
package runtimemodels.chazm.model.id

import com.google.inject.AbstractModule

/**
 * The [IdModule] class provides a Guice binding module for id.
 *
 * @author Christopher Zhong
 * @since 7.0.0
 */
class IdModule : AbstractModule() {

    override fun configure() {
        bind(IdFactory::class.java).to(IdFactoryImpl::class.java)
    }

}
